lib/iex/lib/iex/helpers.ex

```diff
@@ -457,13 +474,22 @@ defmodule IEx.Helpers do
       13
       iex(2)> value
       13
+      iex(3)> import_file "nonexisting.file.ex", optional: true
+      nil
+
   """
-  defmacro import_file(path) when is_binary(path) do
+  defmacro import_file(path, opts \\ [])
+
+  defmacro import_file(path, opts) when is_binary(path) do
+    optional? = Keyword.get(opts, :optional, false)
     path = Path.expand(path)
-    Code.string_to_quoted! File.read!(path), file: path
+
+    if File.exists?(path) or not optional? do
```

Comment:
You want the opposite order in `or` so you don't check if the file exists in case it is required anyway. :)